### PR TITLE
New management command to work with fixtures

### PR DESCRIPTION
**Pull request by max-arnold**
This command is useful to copy single indentical set of imported Geonames data between different servers (for example staging/production environments) and to speedup subsequent imports ~10x times.

Also it can fetch fixtures from arbitrary remote HTTP server or local folder (they can be large, so it is not optimal to store them in VCS). In order to do that, downloader was factored out to separate class.

### CODE REVIEW

**Comment by jpic** [comment]
Should we perhaps have a fixture server under continuous deployment on an openshift free account ?

**Comment by max-arnold** [comment]
For what purpose? To test HTTP fixture downloader under Travis or to distribute daily fixture snapshots?

**Comment by jpic** [comment]
Distributing daily snapshots would be nice.

**Comment by jpic** [comment]
That's maybe a separate issue, ok for merge !

**Comment by max-arnold** [comment]
I think it will be useful to ask more people (mailing list for example) if they need such a server.

**Comment by max-arnold** [comment]
Are you available on IRC or some other messaging platform? I have several other improvements in queue which I'd like to discuss before implementing them.

**Comment by max-arnold** [comment]
Before merging this PR, I'll give @greenday2 some time to add more unit tests.

**Comment by jpic** [comment]
Yeah I"m on #yourlabs@freenode, nickname is_null or jpic.

**Comment by jpic** [comment]
Could  you please rebase ?

**Comment by max-arnold** [comment]
Sure

**Comment by jpic** [comment]
I think it wants this branch to remove test.sh

File: cities_light/downloader.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+class Downloader(object):
+
+    """Data downloader."""
+
+    def download(self, source, destination, logger, force=False):
+        """Data download method."""
+        # Prevent copying itself
+        parsed_source = urlparse(source)
+        if parsed_source.scheme == 'file':
```
Comment (jpic) [inline cities_light/downloader.py]: Oops, seems like we duplicated this for testing ...

https://github.com/yourlabs/django-cities-light/blame/stable/3.x.x/cities_light/geonames.py#L31-L34

So maybe, please be careful when you rebase to remove one of the two implementations of local file download in the code !

**Comment by max-arnold** [comment]
If you plan to push a new release, don't wait for this PR to be merged. We want to add some tests here. BTW, where is the new place to add new test dependencies like `mock` library?

**Comment by jpic** [comment]
Yeah this PR can be in 3.2.1, it will be much easier to test it now that tests run in only a few minutes rather than hoursssss.

Test deps are in `tox.ini[testenv]deps =`

I'm still on IRC if you'd like

**Comment by max-arnold** [comment]
Michael @greenday2 did a great job with refactored Downloader class, new management command and test coverage. We think this PR is ready to be merged.

@jpic James, could you please review the code and approve it for merging?

**Comment by jpic** [comment]
I reviewed it yesterday when I added my +1, really good, keep up the great work !!

**Comment by jpic** [comment]
Gratzzzzz !​

**Pull request merged**